pandas/core/tools/datetimes.py

```diff
@@ -266,24 +268,21 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
                         if errors == 'raise':
                             raise
                         result = arg
-
-        if result is None:
-            assert format is None or infer_datetime_format
-            result, tz_parsed = tslib.array_to_datetime(
-                arg,
-                errors=errors,
-                utc=tz == 'utc',
-                dayfirst=dayfirst,
-                yearfirst=yearfirst,
-                require_iso8601=require_iso8601
-            )
-    except ValueError as e:
-        # Fallback to try to convert datetime objects
-        try:
-            values, tz = conversion.datetime_to_datetime64(arg)
-            return DatetimeIndex._simple_new(values, name=name, tz=tz)
-        except (ValueError, TypeError):
-            raise e
+        except ValueError as e:
+            # Fallback to try to convert datetime objects
+            try:
```

Comment:
cam you be slightly more specific on 'fallback', e.g. when would this be triggered

Reply:
I'll flesh out the comment.  The gist of it is if tz-aware datetime objects are passed and `utc=True` is _not_ passed, then `array_to_datetime` will raise a `ValueError` and `datetime_to_datetime64` will handle this case.

It's a mess and is a big reason why I've been making array_to_datetime PRs: #24006 is an attempt to fix it, but there are some design questions @mroeschke and I are still batting around.